### PR TITLE
PW-563 Add applicationInformation into M2 

### PR DESCRIPTION
**Pull request by cyattilakiss**
**Description**
Add applicationInformation info for the Client constructor so we can
send it with every request

### CODE REVIEW

File: Gateway/Http/Client/TransactionAuthorization.php [hunk 1]
```diff
@@ -69,6 +69,13 @@ public function __construct(
         $client->setUsername($webserviceUsername);
         $client->setPassword($webservicePassword);
 
+		$client->setAdyenPaymentSource($this->_adyenHelper->getModuleName(), $this->_adyenHelper->getModuleVersion());
```
Comment (lancergr) [inline Gateway/Http/Client/TransactionAuthorization.php]: let's move it to helper as factory method

**Comment by lancergr** [comment]
let's also add the new version dependency on composer.json https://github.com/Adyen/adyen-magento2/blob/develop/composer.json#L17

**Comment by rikterbeek** [comment]
> let's also add the new version dependency on composer.json https://github.com/Adyen/adyen-magento2/blob/develop/composer.json#L17

agree after release make it this release or higher so that it will keep taking the latest version

File: Gateway/Http/Client/TransactionAuthorization.php [hunk 2]
```diff
@@ -69,6 +69,13 @@ public function __construct(
...
+
+		$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+		$productMetadata = $objectManager->get('Magento\Framework\App\ProductMetadataInterface');
+
```
Comment (rikterbeek) [inline Gateway/Http/Client/TransactionAuthorization.php]: Let's not use objectManager this is a old deprecated way. Best to use dependency injection in the constructor.

File: Gateway/Http/Client/TransactionCancel.php [hunk 3]
```diff
@@ -64,6 +64,12 @@ public function __construct(
         $client->setUsername($webserviceUsername);
         $client->setPassword($webservicePassword);
 
+		$client->setAdyenPaymentSource($this->_adyenHelper->getModuleName(), $this->_adyenHelper->getModuleVersion());
+
+		$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+		$productMetadata = $objectManager->get('Magento\Framework\App\ProductMetadataInterface');
```
Comment (rikterbeek) [inline Gateway/Http/Client/TransactionCancel.php]: Let's not use objectManager, best to use dependency injection in the constructor.

File: Gateway/Http/Client/TransactionCapture.php [hunk 4]
```diff
@@ -64,6 +64,12 @@ public function __construct(
         $client->setUsername($webserviceUsername);
         $client->setPassword($webservicePassword);
 
+		$client->setAdyenPaymentSource($this->_adyenHelper->getModuleName(), $this->_adyenHelper->getModuleVersion());
+
+		$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+		$productMetadata = $objectManager->get('Magento\Framework\App\ProductMetadataInterface');
```
Comment (rikterbeek) [inline Gateway/Http/Client/TransactionCapture.php]: Let's not use objectManager, best to use dependency injection in the constructor.

File: Gateway/Http/Client/TransactionRefund.php [hunk 5]
```diff
@@ -64,6 +64,12 @@ public function __construct(
         $client->setUsername($webserviceUsername);
         $client->setPassword($webservicePassword);
 
+        $client->setAdyenPaymentSource($this->_adyenHelper->getModuleName(), $this->_adyenHelper->getModuleVersion());
+
+		$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+		$productMetadata = $objectManager->get('Magento\Framework\App\ProductMetadataInterface');
+		$client->setExternalPlatform($productMetadata->getName(), $productMetadata->getVersion());
```
Comment (rikterbeek) [inline Gateway/Http/Client/TransactionRefund.php]: Let's not use objectManager, best to use dependency injection in the constructor.

File: Model/Api/PaymentRequest.php [hunk 6]
```diff
@@ -88,6 +88,12 @@ private function createClient($storeId) {
         $client->setUsername($webserviceUsername);
         $client->setPassword($webservicePassword);
 
+        $client->setAdyenPaymentSource($this->_adyenHelper->getModuleName(), $this->_adyenHelper->getModuleVersion());
+
+		$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+		$productMetadata = $objectManager->get('Magento\Framework\App\ProductMetadataInterface');
+		$client->setExternalPlatform($productMetadata->getName(), $productMetadata->getVersion());
```
Comment (rikterbeek) [inline Model/Api/PaymentRequest.php]: Let's not use objectManager, best to use dependency injection in the constructor.

File: Helper/Data.php [hunk 7]
```diff
@@ -1131,4 +1163,37 @@ public function getOpenInvoiceLineData(
...
+	 *
+	 * @param $storeId
+	 * @return \Adyen\Client
+	 * @throws \Adyen\AdyenException
+	 */
+    public function initializeAdyenClient($storeId = null)
+	{
+		// initialize client
+		$webserviceUsername = $this->getWsUsername();
+		$webservicePassword = $this->getWsPassword();
```
Comment (Aleffio) [inline Helper/Data.php]: let's use $this->getWsUsername($storeId)
and $this->getWsPassword($storeId)

**Pull request merged**